src/coreclr/tools/dotnet-pgo/Program.cs

```diff
@@ -256,11 +256,11 @@ public int Run()
             {
                 return InnerDumpMain();
             }
-            if (_inputFilesToMerge != null)
+            if (_inputFilesToMerge.Count > 0)
             {
                 return InnerMergeMain();
             }
-            if (_inputFilesToCompare != null)
+            if (_inputFilesToCompare.Length > 0)
```

Comment:
I thought it uses `default(T)` when user defined default value is not specified:
https://github.com/dotnet/runtime/blob/95d36a9a0228ad8690cfbad6453ea5bfa90de75e/src/coreclr/tools/dotnet-pgo/PgoRootCommand.cs#L20-L21

Apparently not? 🤔

`_inputFilesToMerge` also does not _run_ the `BuildPathList` helper when option is not specified (third argument is `false`).

Reply:
Apparently `GetValueForOption` ends up returning empty list/array in this case.

Reply:
Ah `GetValueForOption` could be the reason. I'll double-check if it happens with non-collection types. (I _think_ with `Option<string>` it returns `null`, `Option<int>` is `0` and with `Option<bool>`, `false`). Need to revisit all root commands. 😁

Reply:
We've had this issue in a few places now when switching from System.CommandLine v1 to v2, so I guess it was a breaking change there. E.g. #69656 and #76820 was the same problem. I actually did audit these after #76820, but I guess I missed this one.